oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,246 @@
+
+Library Repositories
+--------------------
+
+All Open edX library repositories should contain a CHANGELOG.rst.
+
+Frontend Libraries
+~~~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, frontend libraries should also include a CHANGELOG.rst. However, as you'll see below, this file will simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+  The release notes format should include information recommended in `OEP-44: Changelogs`_.
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _`OEP-44: Changelogs`: https://open-edx-proposals.readthedocs.io/en/latest/oep-0044-bp-changelogs.html
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Every package published to npm should have a corresponding changelog entry for the new version.
+
+All new frontend libraries have been using an npm package called `semantic-release`_, which automates publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_. The versions are automatically tagged using semantic versioning, as detailed in `OEP-47: Semantic Versioning`_.
+
+Please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+  The release notes format should include information recommended in `OEP-44: Changelogs`_.
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _`OEP-44: Changelogs`: https://open-edx-proposals.readthedocs.io/en/latest/oep-0044-bp-changelogs.html
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Every package published to PyPI should have a corresponding changelog entry for the new version.
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time to avoid issues where the CHANGELOG.rst and GitHub Releases are not in sync.
```

Comment:
Without external tooling, GitHub Releases are painful to use.  They are pre-populated with only the last commit message on the branch.  They are an extra editing step after the tag has been created, with no possibility for review.

I would think to reduce developer effort, and avoid losing information, the best recommendation would be to manually edit a CHANGELOG.rst file.

Reply:
I haven't found it painful to use, but started recommending against it and provided some reasons.  I'm on board.